aptos-move/aptos-vm/src/keyless_validation.rs

```diff
@@ -47,9 +47,24 @@ fn get_resource_on_chain_at_addr<T: MoveStructType + for<'a> Deserialize<'a>>(
     module_storage: &impl ModuleStorage,
 ) -> anyhow::Result<T, VMStatus> {
     let struct_tag = T::struct_tag();
+    if !struct_tag.address.is_special() {
```

Comment:
where do we ensure the address is also special? for example I'm looking at https://github.com/aptos-labs/aptos-core/blob/7c61dc34c9b82dabcb8faf7ec1b005d93c57e89e/aptos-move/aptos-vm/src/keyless_validation.rs#L89

Reply:
It is implicitly enforced by trait bound on tag:
```
pub trait MoveStructType {
    const ADDRESS: AccountAddress = crate::language_storage::CORE_CODE_ADDRESS;
    ...
```
so when we look up metadata for this one we always have 0x1